api/types.go

```diff
@@ -523,13 +523,16 @@ type Pin struct {
 	Allocations          []peer.ID
 	ReplicationFactorMin int
 	ReplicationFactorMax int
+	Recursive            bool
```

Comment:
This is going to require a migration

Reply:
@hsanjuan What determines when a migration is required?

Reply:
@lanzafame if the state format changes so that updated versions of cluster will treat old state data in an incorrect way or be unable to directly deserialize old state files into the new format when loading then a migration is needed.  In this case a migration is needed because the state format (e.g. [MapState](https://github.com/ipfs/ipfs-cluster/blob/master/state/mapstate/map_state.go#L28)) uses PinSerial and the because the default value of booleans in golang is false while the default value of `ipfs pin`'s recursive argument is true.